### PR TITLE
Added cross validation

### PR DESCRIPTION
**Pull request by mstechly**
I added possibility of defining a cross validation set and monitoring how it performs during the training.
I also checked if it doesn't make the computation longer without defining CV set. It doesn't.

It must be calculated and assigned in the fitness function the same way as for training set.

### CODE REVIEW

**Comment by CodeReclaimers** [comment]
Thanks, and sorry for not responding earlier!  I'll have a look and get this added over the weekend.

**Comment by mstechly** [comment]
Hi. It would be nice if you accepted or rejected it ;)

**Comment by CodeReclaimers** [comment]
Apologies--I've been swamped with a bunch of other stuff lately, I will take care of it this weekend. :)
